### PR TITLE
Machiliv/isc bind

### PR DESCRIPTION
**Pull request by mkchiliveri**
Required items, please complete
   
   Change(s):
   - ISC Bind Solution

   Reason for Change(s):
   - New solution

   Version Updated:
   - NA

   Testing Completed:
   - Yes

### CODE REVIEW

**Comment by v-sabiraj** [comment]
Hey @mkchiliveri, can you please move the sample data file to [correct folder](https://github.com/Azure/Azure-Sentinel/tree/master/Sample%20Data/Syslog)?

**Comment by v-sabiraj** [comment]
Please update the sample data file name same as query, i.e ISCBind.csv rather than ISCBIND.csv

**Comment by oshezaf** [comment]
@mkchiliveri please do not merge with the sample file. I will discuss with you offline.

File: Solutions/ISC Bind/Data Connectors/Connector_Syslog_ISCBind.json [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+          "type": "InstallAgent"
+        }
+      ]
+    },
+    {
+      "title": "3. Configure and connect the ISC Bind",
+      "description": "1. Follow these instructions to configure the ISC Bind to forward syslog: \n - [DNS Logs](https://kb.isc.org/docs/aa-01526) \n2. Configure Syslog to send the Syslog traffic to Agent. Use the IP address or hostname for the Linux device with the Linux agent installed as the Destination IP address."
+    }
+  ]
+}
```
Comment (v-sabiraj) [inline Solutions/ISC Bind/Data Connectors/Connector_Syslog_ISCBind.json:124]: @mkchiliveri, Can you please check if metadata needs to be added?, thanks.

File: Solutions/ISC Bind/Parsers/ISCBind.txt [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        // -- multi-line mode (?m) needed in regex since samples included new lines. Might not exist in real logs
+        extract = extract_all(@"^(\w+) ([\w\+\-]+?)(?: |$)(?m)(.*?$)?", rest_of_string)[0]
+| extend 
+        DnsResponseCodeName = tostring(extract[0]),
+        DnsFlags = tostring(extract[1]),
+        DnsResponseName = tostring(extract[2])
+| extend 
+        EventSubType = "response"
+| project-away extract, rest_of_string,SyslogMessage;
+
```
Comment (devikamehra) [inline Solutions/ISC Bind/Parsers/ISCBind.txt:71]: There shouldn't be any spaces in between.

File: Solutions/ISC Bind/Parsers/ISCBind.txt [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        DnsQuery:string " "
+        DnsQueryClassName:string " "
+        DnsQueryTypeName:string " "
+        DnsFlags:string
+| extend DnsQuery = replace_string (DnsQuery, '@@@', ' ')
+| extend 
+        EventSubType = "request",
+        DnsResponseCodeName = "NA"
+| project-away SyslogMessage, ProcessName, ProcessID, Facility, SeverityLevel, HostName;
+
```
Comment (devikamehra) [inline Solutions/ISC Bind/Parsers/ISCBind.txt:46]: Remove the spaces

File: Solutions/ISC Bind/Parsers/ISCBind.txt [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+//    It is recommended to name the Function Alias, as ISCBind
+// 4. Kusto Functions can typically take up to 15 minutes to activate. You can then use Function Alias for other queries.
+//
+// REFERENCES: 
+// Using functions in Azure monitor log queries: https://docs.microsoft.com/azure/azure-monitor/log-query/functions
+// 
+// LOG SAMPLES:
+// This parser assumes the raw log are formatted as follows:
+//
+// <190>Jun 30 16:15:59 218.254.206.240 named[22442]: client @0a0a00000a0a00 247.69.221.212#35306 (abc-exam.d.example.com): view 7: query: test123.example.com IN A -E(0)DCK (218.254.206.240)
```
Comment (devikamehra) [inline Solutions/ISC Bind/Parsers/ISCBind.txt:22]: Remove the data from here or mask it.

File: Solutions/ISC Bind/Parsers/ISCBind.txt [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+// <190>Jun 30 16:15:59 218.254.206.240 named[22442]: 30-Jun-2022 16:15:59.366 client @0a0a00000a0a00 247.69.221.212#35306 (abc-exam.d.example.com): view 7: UDP: query: test.example.com IN A response: NXDOMAIN -AE
+
+let request = Syslog
```
Comment (devikamehra) [inline Solutions/ISC Bind/Parsers/ISCBind.txt:25]: Parser is not working for all the inputs.

**Review by devikamehra — CHANGES_REQUESTED** [review]

File: Solutions/ISC Bind/Parsers/ISCBind.txt (hunk 2 above)
Comment (mkchiliveri) [reply Solutions/ISC Bind/Parsers/ISCBind.txt:83]: Removed spaces.

File: Solutions/ISC Bind/Parsers/ISCBind.txt (hunk 3 above)
Comment (mkchiliveri) [reply Solutions/ISC Bind/Parsers/ISCBind.txt:59]: Removed the spaces.

File: Solutions/ISC Bind/Parsers/ISCBind.txt (hunk 4 above)
Comment (mkchiliveri) [reply Solutions/ISC Bind/Parsers/ISCBind.txt]: Removed the data

File: Solutions/ISC Bind/Parsers/ISCBind.txt (hunk 5 above)
Comment (mkchiliveri) [reply Solutions/ISC Bind/Parsers/ISCBind.txt:19]: Updated the parser.

File: Solutions/ISC Bind/Parsers/ISCBind.txt (hunk 3 above)
Comment (devikamehra) [reply Solutions/ISC Bind/Parsers/ISCBind.txt:59]: Not removed

File: Solutions/ISC Bind/Parsers/ISCBind.txt [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+| extend 
+        EventSubType = "request",
+        DnsResponseCodeName = "NA"
+| project-away SyslogMessage,ProcessName,ProcessID,Facility,SeverityLevel,HostName;
+
+
+let requestcache = Syslog 
+| where SyslogMessage has_all ("client", "query (cache)") and SyslogMessage !has "response:"
+    // -- trim_end needed to remove optional repetative IP address at the end.
+| extend SyslogMessage = trim_end (@' \([\d\.]+\)', SyslogMessage)
```
Comment (devikamehra) [inline Solutions/ISC Bind/Parsers/ISCBind.txt:43]: We can remove this as request cache does not have the server IP. Also, the next line with replace.

File: Solutions/ISC Bind/Parsers/ISCBind.txt [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+| parse SyslogMessage with
+        * "client " * " "
+        SrcIpAddr:string "#" 
+        SrcPortNumber:int *
+        "query (cache) '"
+        DnsQuery:string "/"
+        DnsQueryTypeName:string "/"
+        DnsQueryClassName:string "' "
+        Action
+| extend DnsQuery = replace_string (DnsQuery, '@@@', ' ')
```
Comment (devikamehra) [inline Solutions/ISC Bind/Parsers/ISCBind.txt:54]: Remove this also

File: Solutions/ISC Bind/Parsers/ISCBind.txt [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+//    It is recommended to name the Function Alias, as ISCBind
+// 4. Kusto Functions can typically take up to 15 minutes to activate. You can then use Function Alias for other queries.
+//
+// REFERENCES: 
+// Using functions in Azure monitor log queries: https://docs.microsoft.com/azure/azure-monitor/log-query/functions
+
+let request = Syslog 
+| where SyslogMessage has_all ("client", "query:") and SyslogMessage !has "response:"
+    // -- trim_end needed to remove optional repetative IP address at the end.
+| extend SyslogMessage = trim_end (@' \([\d\.]+\)', SyslogMessage)
```
Comment (devikamehra) [inline Solutions/ISC Bind/Parsers/ISCBind.txt:22]: Remove this and replace it with ServerIpAddr in Parser

**Review by devikamehra — CHANGES_REQUESTED** [review]

File: Solutions/ISC Bind/Parsers/ISCBind.txt (hunk 3 above)
Comment (mkchiliveri) [reply Solutions/ISC Bind/Parsers/ISCBind.txt]: Removed the empty lines.

File: Solutions/ISC Bind/Parsers/ISCBind.txt (hunk 7 above)
Comment (mkchiliveri) [reply Solutions/ISC Bind/Parsers/ISCBind.txt]: Removed

File: Solutions/ISC Bind/Parsers/ISCBind.txt (hunk 8 above)
Comment (mkchiliveri) [reply Solutions/ISC Bind/Parsers/ISCBind.txt]: Removed and added Server IP address

**Comment by v-atulyadav** [comment]
Hi @devikamehra , can you please check the changes done by @mkchiliveri and provide feedback on this. Thanks.

**Review by devikamehra — APPROVED** [review]

**Pull request merged**